azurerm/data_source_image.go

```diff
@@ -12,9 +17,23 @@ func dataSourceArmImage() *schema.Resource {
 		Read: dataSourceArmImageRead,
 		Schema: map[string]*schema.Schema{
 
+			"name_regex": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ForceNew:      true,
+				ValidateFunc:  validation.ValidateRegexp,
+				ConflictsWith: []string{"name"},
+			},
+			"name_descending": {
```

Comment:
after chatting with @radeksimko - can we rename this `sort_descending`? it's possible we'll filter by additional fields in the future